### PR TITLE
Make ranklist case insensitive.

### LINKED ISSUE #278: make ranklist case insensitive

### PR DESCRIPTION
**Pull request by aryanc403**
Fixes #278

### CODE REVIEW

**Comment by aryanc403** [comment]
Updated with custom case insensitive dictionary instead of relying on request's internal stuff. As directed by @algmyr

File: tle/util/handledict.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""
+    A case insensitive dictionay with bare minimum functions required for handling usernames.
+"""
+
+class HandleDict:
+    def __init__(self):
+        self._store = {}
+
+    def getlower(self,key):
+        return key.lower() if type(key)==str else key
```
[the remark this answers is not in the document]
Comment (aryanc403) [reply tle/util/handledict.py:10]: Team ids are integers and usernames are string. Earlier I had to put something similar to make it a string.
```
id_ = "Team:"+str(row.party.teamId) if row.party.teamId else row.party.members[0].handle
```

**Comment by aryanc403** [comment]
Fixed.

File: tle/util/handledict.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (algmyr) [reply tle/util/handledict.py:11]: Ok, cool.

**Pull request merged**